### DISCUSSION TITLE
HFS 2.3 Windows vs HFS 3 Linux

### DISCUSSION THREAD

**Discussion by quite68** [rejetto/hfs#150] [Q&A] [answered]
I'm currently using HFS2.3 for Windows. I'm looking at HFS3 for Linux.

The curl commands for upload are different for these two.

If I have a dumb client that doesn't know what it's talking to and just does wget and curl, how can it tell which curl command to use for upload?

Is it possible for the client to determine the version of HFS?
Or can it download some log or settings or similar?

**Comment by rejetto** [comment, marked as answer]
is curl command you use on hfs2 not compatible with hfs3? i can't remember.
hfs3 supports both forms:
- curl -T file url/
- curl -F upload=@file url/

to know the version you can make a request before and look at the "server" header.
HFS3 is providing the server header only if you request a folder, for example the root /

**Comment by rejetto** [comment]
let me know what kind of project you are making with this
